### DISCUSSION TITLE
Local Storage

### DISCUSSION THREAD

**Discussion by AmjedXKhattab** [neuefische/questions#800] [Web: Self-directed learning] [answered]
const [reviews, setReviews] = useLocalStorageState(`reviews_${card.id}`);

  const onAddReview = (cardId, review) => {
    const updatedReviews = { ...reviews, [cardId]: review };
    setReviews(updatedReviews); };
  
  Ich habe ein Problem mit diesem Code. Ich habe das lokale Speicherpaket installiert und auch einen Import für den lokalen Speicher durchgeführt, aber trotz Umsetzung aller Anforderungen erscheint ein Fehler bezüglich des lokalen Speichers.
  
  Unhandled Runtime Error
TypeError: (0,use_local_storage_state__WEBPACK_IMPORTED_MODULE_5__.useLocalStorageState) is not a function. (In '(0,use_local_storage_state__WEBPACK_IMPORTED_MODULE_5__.useLocalStorageState)("reviews_".concat(card.id))', '(0,use_local_storage_state__WEBPACK_IMPORTED_MODULE_5__.useLocalStorageState)' is undefined)

**Comment by doemser** [comment, marked as answer]
Looks like you need to install the package first, or you havent imported the hook:
```jsx
import useLocalStorageState from "use-local-storage-state";
```
```
npm i use-local-storage-state
```

**Reply by AmjedXKhattab** [reply]
@doemser  I have installed the package.  I also did an import but the problem is the same

**Reply by AmjedXKhattab** [reply]
@doemser  thank you , the problem has been resolved

**Comment by mbosselmann** [comment]
Hi Amjed,
can you also add a link to the branch with the current state so I can check? :)

**Reply by AmjedXKhattab** [reply]
@mbosselmann  hello,
I discovered the error, I was using an old library package installation that was not compatible with the system, so it did not work, but Dominic suggested another compatible one and the problem was solved.
thanks for help

**Reply by mbosselmann** [reply]
Wonderful! 👍
